src/auth-service/utils/request.js

```diff
@@ -54,40 +43,38 @@ const request = {
       //   }
       // });
 
-      let filterRequestBody = {};
-      filterRequestBody["body"] = {};
-      filterRequestBody["query"] = {};
-      filterRequestBody["body"] = req;
-      filterRequestBody["query"]["category"] = category;
+      let filter = { email };
 
-      const responseFromGenerateCandidateFilter =
-        generateFilter.candidates(filterRequestBody);
-
-      if (responseFromGenerateCandidateFilter.success === false) {
-        callback({
-          success: false,
-          message: "internal server error",
-          errors: { message: responseFromGenerateCandidateFilter.error },
-        });
-      }
-
-      let filter = responseFromGenerateCandidateFilter.data;
-      const responseFromListCandidates = await request.list({
-        tenant,
+      const responseFromListCandidates = await CandidateModel(
+        tenant.toLowerCase()
+      ).list({
         filter,
       });
 
-      if (responseFromListCandidates.success === true) {
-        if (
-          Array.isArray(responseFromListCandidates.data) &&
-          responseFromListCandidates.data.length === 1
-        ) {
+      logObject("responseFromListCandidates", responseFromListCandidates);
+
+      if (
+        responseFromListCandidates.message ===
+        "successfully listed the candidates"
+      ) {
+        callback({
+          success: true,
+          message: "candidate already exists",
+          status: httpStatus.OK,
+        });
+      } else if (responseFromListCandidates.message === "no candidates exist") {
+        const responseFromListUsers = await UserModel(
+          tenant.toLowerCase()
+        ).list({
+          filter,
+        });
+        if (responseFromListUsers.message === "successfully listed the users") {
           callback({
             success: true,
-            message: "candidate already exists or successfully created",
+            message: "candidate already exists as a user",
             status: httpStatus.OK,
           });
-        } else {
+        } else if (responseFromListUsers.message === "no users exist") {
           const responseFromCreateCandidate = await CandidateModel(
```

Comment:
On create candidate if there is no corresponding User or Candidate account with those details.